tests/test_ec2.py

```diff
@@ -466,3 +469,113 @@ def test_list_clusters(ec2_patch, monkeypatch):
     mocked_eks.clusters_list = {'clusters': ['hastags', 'ignored']}
     all_clusters = ec2_patch.all_clusters()
     assert all_clusters == {'region1': ['hastags']}
+
+
+class MockedKubernetesConfig():
+    def load_kube_config(self, *args, **kwargs):
+        return True
+
+
+class MockedKubernetesClient():
+    def __init__(self, jobs=[]):
+        self.jobs = jobs
+        self.deleted_jobs = []
+
+    # pylint: disable=C0103
+    def BatchV1Api(self):
+        return self
+
+    def list_job_for_all_namespaces(self, *args, **kwargs):
+        return MockedKubernetesResult(self.jobs)
+
+    def delete_namespaced_job(self, name, namespace):
+        self.deleted_jobs.append(name)
+
+
+class MockedKubernetesResult():
+    def __init__(self, items):
+        self.items = items
+
+
+class MockedKubernetesJobStatus():
+    def __init__(self, age):
+        self.start_time = datetime.now(timezone.utc) - timedelta(days=age)
+
+
+class MockedKubernetesJobMetadata():
+    def __init__(self, name):
+        self.name = name
+        self.namespace = "default"
+
+
+class MockedKubernetesJob():
+    def __init__(self, name, age):
+        self.status = MockedKubernetesJobStatus(age)
+        self.metadata = MockedKubernetesJobMetadata(name)
+
+
+def cmd_exec_exception(object, cmd):
+    raise subprocess.CalledProcessError(1, cmd)
+
+
+@pytest.fixture
+def k8s_patch(monkeypatch):
+    monkeypatch.setattr(EC2, 'check_credentials', lambda *args, **kwargs: True)
+    monkeypatch.setattr(Provider, 'read_auth_json', lambda *args, **kwargs: '{}')
+    monkeypatch.setattr(EC2, 'get_all_regions', lambda self: ['region1'])
+    monkeypatch.setattr(PCWConfig, 'get_feature_property', mock_get_feature_property)
+
+    return EC2('fake')
+
+
+def test_kubectl_client(k8s_patch, monkeypatch):
+    monkeypatch.setattr(Provider, 'cmd_exec', lambda *args, **kwargs: 1)
+    monkeypatch.setattr(kubernetes, 'config', MockedKubernetesConfig())
+    mocked_client1 = MockedKubernetesClient(1)
+    monkeypatch.setattr(kubernetes, 'client', mocked_client1)
+    assert k8s_patch.kubectl_client("region1", "cluster") == mocked_client1
```

Comment:
I would also assert size of `k8s_patch.__kubectl_client` if it is possible ( not sure due to `__` at the beginning python may not allow it to you )